### PR TITLE
Add objects to emailtempl

### PR DESCRIPTION
**Pull request by smirnovaka**
Tested. 
All emailTemplates reviewed for use of object.propertyName. 
Updated EmailPropertyService to set objects where appropriate.

### CODE REVIEW

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 1]
```diff
@@ -54,26 +58,44 @@ foam.CLASS({
...
-            if ( value == null ) {
-              value = "";
-              foam.nanos.logger.StdoutLogger.instance().warning("No value provided for variable",v);
-            }
-            ((StringBuilder) x.get("sb")).append(value);
-            return value;
-          }
-        };
-        grammar.addAction("SIMPLE_VAL", simpleValAction);
+grammar.addSymbol("VALUE", new Seq2(2,3,Literal.create("{{"), Whitespace.instance(), grammar.sym("WORD"), new Optional(
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: leading spacing missing.

**Review by jlhughes — COMMENTED** [review]
So this allows
args.put("account", account);
and later
{{account.some_property}}
?

**Comment by smirnovaka** [comment]
> So this allows args.put("account", account); and later {{account.some_property}} ?

yes

**Comment by kgrgreer** [comment]
I don't understand, I thought we already supported this ;-).

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 2]
```diff
@@ -54,26 +58,44 @@ foam.CLASS({
...
+    if ( vals.length > 1 && vals[1] == null ) {
+      String value = (String) ((Map) x.get("values")).get(v);
+      if ( value == null ) {
+        value = "";
+        foam.nanos.logger.StdoutLogger.instance().warning("No value provided for variable",v);
+      }
+      ((StringBuilder) x.get("sb")).append(value);
+      return value;
+    }
+     if ( ((Map) x.get("values")).get(v) == null ) throw new RuntimeException("Object is not provided " + v);
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: indentation is off

**Comment by kgrgreer** [comment]
LGTM, +/- some questionable indentation

**Comment by jlhughes** [comment]
@penzital what will happen with existing {{alarm.note}} for example?

**Comment by smirnovaka** [comment]
> @penzital what will happen with existing {{alarm.note}} for example?

still should work

**Review by jlhughes — APPROVED** [review]

**Pull request merged**